test/integration/concrete/lockup/burn/burn.t.sol

```diff
@@ -148,16 +149,58 @@ abstract contract Burn_Integration_Concrete_Test is Integration_Test, Lockup_Int
         lockup.getRecipient(streamId);
     }
 
-    function test_Burn_CallerNFTOwner()
+    function test_Burn_CallerNFTOwner_TransferableNFT()
         external
         whenNotDelegateCalled
         givenNotNull
-        givenStreamHasBeenDepleted
+        givenStreamHasBeenDepleted(streamId)
         whenCallerAuthorized
         givenNFTExists
     {
         lockup.burn(streamId);
         vm.expectRevert("ERC721: invalid token ID");
         lockup.getRecipient(streamId);
     }
+
+    modifier givenNonTransferableNFT() {
+        noTransferStreamId = createDefaultStreamNotTransferable();
+        _;
+    }
+
+    function test_Burn_CallerApprovedOperator_NonTransferableNFT()
+        external
+        givenNonTransferableNFT
+        whenNotDelegateCalled
+        givenNotNull
+        givenStreamHasBeenDepleted(noTransferStreamId)
+        whenCallerAuthorized
+        givenNFTExists
```

Comment:
The `givenNonTransferableNFT` modifier should have been defined after `givenNotNull`.